NitroxLauncher/Models/Utils/WindowsHelper.cs

```diff
@@ -55,6 +64,100 @@ internal static void RestartAsAdmin()
+            {
+                MessageBox.Show("There was a problem configuring the Windows firewall. Try restarting the launcher as administrator or manually adding firewall rules for Nitrox programs.");
+            }
+        }
+
+        private static void CheckServerFirewallRules()
+        {
+            string serverRuleName = "nitroxserver-subnautica.exe";
+            string serverPath = Path.Combine(Path.GetDirectoryName(Assembly.GetEntryAssembly().Location), ServerLogic.SERVER_EXECUTABLE);
+
+            AddExclusiveFirewallRule(serverRuleName, serverPath);
+        }
+
+        private static void CheckClientFirewallRules()
+        {
+            string clientRuleName = "Subnautica";
+            string clientPath = Path.Combine(LauncherLogic.Config.SubnauticaPath, "Subnautica.exe");
+
+            AddExclusiveFirewallRule(clientRuleName, clientPath);
+        }
+
+        private static void CheckHamachiFirewallRules()
+        {
+            static IPAddress GetHamachiAddress()
+            {
+                foreach (NetworkInterface networkInterface in NetworkInterface.GetAllNetworkInterfaces())
+                {
+                    if (networkInterface.Name == "Hamachi")
+                    {
+                        foreach (UnicastIPAddressInformation addressInfo in networkInterface.GetIPProperties().UnicastAddresses)
+                        {
+                            if (addressInfo.Address.AddressFamily == AddressFamily.InterNetwork)
+                            {
+                                return addressInfo.Address;
+                            }
+                        }
+                    }
+                }
+
+                return null;
+            }
+
+            if (GetHamachiAddress() == null || !FirewallWASRule.IsLocallySupported)
+            {
+                return;
+            }
+
+            foreach (IFirewallRule firewallRule in FirewallManager.Instance.Rules)
+            {
+                if (firewallRule.Name == "Hamachi")
+                {
+                    return;
+                }
+            }
+
+            FirewallWASRule rule = new("Hamachi", FirewallAction.Allow, FirewallDirection.Inbound, FirewallManager.Instance.GetActiveProfile().Type)
+            {
+                Description = "Ignore Hamachi network",
+                Protocol = FirewallProtocol.Any,
+                RemoteAddresses = new[] { new NetworkAddress(GetHamachiAddress()) },
+                IsEnable = true
+            };
+
+            FirewallManager.Instance.Rules.Add(rule);
+        }
+
+        private static void AddExclusiveFirewallRule(string name, string filePath)
+        {
+            if (!FirewallRuleExists(name))
+            {
+                AddFirewallRule(name, filePath);
+            }
+        }
+
+        private static bool FirewallRuleExists(string name) => FirewallManager.Instance.Rules.Any(rule => rule.FriendlyName == name);
```

Comment:
This check ignores the direction (inbound or outbound) so an existing outbound rule for Subnautica would mean that no inbound rule is created.

Reply:
Yeah maybe we shouldn't be checking firewall rules at any startup since some firewalls don't need rules like this to work properly with Nitrox so we would end up adding useless rules. Adding an optional way of checking through the troubleshoot page seems fair